### PR TITLE
Working on drone/drone#959

### PR DESCRIPTION
**Pull request by benschumacher**
Cleaning up build traces of repo when it is deleted. Some of the cleanup steps may not be necessary for buckets that have been rolled up into the main build document.

### CODE REVIEW

**Comment by benschumacher** [comment]
Just FYI. I created a separate repo_del_test.go to avoid stepping on @oliveiradan's efforts for repo_test.go. These should be merged together.

File: datastore/bolt/repo.go [hunk 1]
```diff
@@ -132,23 +132,43 @@ func (db *DB) SetRepoKeypair(repo string, keypair *common.Keypair) error {
...
+
+	return db.Update(func(t *bolt.Tx) error {
+		err := t.Bucket(bucketRepo).Delete(key)
+		if err != nil {
+			return err
+		}
+		t.Bucket(bucketRepoKeys).Delete(key)
+		t.Bucket(bucketRepoParams).Delete(key)
+
+		// should we just ignore these error conditions? or should
```
Comment (benschumacher) [inline datastore/bolt/repo.go]: Whoops. I meant to clean this up a bit more, but how should we handle errors. In practice, I think BoltDB is unlikely to fail through this code path, but I wasn't sure if it was reasonable to rollback the entire "delete" operation if one part hiccuped.

File: datastore/bolt/repo.go [hunk 2]
```diff
@@ -132,23 +132,43 @@ func (db *DB) SetRepoKeypair(repo string, keypair *common.Keypair) error {
...
-	t.Bucket(bucketRepoParams).Delete(key)
-	// TODO(bradrydzewski) delete all builds
-	// TODO(bradrydzewski) delete all tasks
-	return t.Commit()
+	})
+}
+
+// deleteTracesOfRepo cleans up build leftovers when a repo is removed
+func (db *DB) deleteTracesOfRepo(t *bolt.Tx, repoKey []byte) error {
+	err := error(nil)
```
Comment (benschumacher) [inline datastore/bolt/repo.go]: re: Note above, this is a particularly egregious waste of code.

**Comment by bradrydzewski** [comment]
great, we'll merge those test files in a follow-up

File: datastore/bolt/repo.go (hunk 1 above)
Comment (bradrydzewski) [inline datastore/bolt/repo.go]: I think we'll want to handle all errors (even though my initial code did not). I'm cool addressing this now, or coming back in a few weeks and making an effort to cleanup / harden the codebase

**Pull request merged**